src/python/pants/engine/README.md

```diff
@@ -168,44 +182,32 @@ To compute a value for a Node, the engine uses the `Node.run` method starting fr
 roots. If a Node needs more inputs, it requests them via `Context.get`, which will declare a
 dependency, and memoize the computation represented by the requested `Node`.
 
-The initial Nodes are [launched by the engine](https://github.com/pantsbuild/pants/blob/16d43a06ba3751e22fdc7f69f009faeb59a33930/src/rust/engine/src/scheduler.rs#L116-L126),
-but the rest of execution is driven by Nodes recursively calling `Context.get` to request their
-dependencies.
+This recorded `Graph` tracks all dependencies between `@rules` and builtin "intrinsic" rules that
+provide filesystem and network access. That dependency tracking allows for invalidation and dirtying
+of `Nodes` as their dependencies change.
 
-### Registering Rules
+## Registering Rules
 
-Currently, it is only possible to load rules into the pants scheduler in two ways: by importing and
-using them in `src/python/pants/bin/engine_initializer.py`, or by adding them to the list returned
-by a `rules()` method defined in `src/python/backend/<backend_name>/register.py`. Plugins cannot add
-new rules yet. Unit tests, however, can mix in `TestBase` from
-`tests/python/pants_test/test_base.py` to generate and execute a scheduler from a given set of
-rules.
+The recommended way to install `@rules` is to return them as a list from a `def rules()` definition
+in a plugin's `register.py` file. Unit tests can either invoke `@rules` with fully mocked
```

Comment:
Short example would be helpful of what the `def rules()` looks like. You don't need to show the implementation of the rules, just the signatures. Something like this.

```python
class List:

  @rule(List, [Console, List.Options, Specs])
  def list_targets(console, list_options, specs):
    ...
  
  def rules():
    return [
      list_targets,
    ]
```

I'd think of this example like an ~integration example. You showed above how the building blocks work like `datatype` and `@rule`. Here, you show how it all comes together.